src/os/vxworks/src/vxworks7/os-impl-common.c

```diff
@@ -0,0 +1,79 @@
+/*
+ *  NASA Docket No. GSC-18,370-1, and identified as "Operating System Abstraction Layer"
+ *
+ *  Copyright (c) 2019 United States Government as represented by
+ *  the Administrator of the National Aeronautics and Space Administration.
+ *  All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * \file     os-impl-common.c
+ * \ingroup  vxworks
+ * \author   joseph.p.hickey@nasa.gov
+ *
+ */
+/****************************************************************************************
+                                    INCLUDE FILES
+****************************************************************************************/
+
+#include "os-vxworks.h"
+
+#include "os-shared-common.h"
+
+#include <errnoLib.h>
+#include <objLib.h>
+#include <semLib.h>
+#include <sysLib.h>
+#include <taskLib.h>
+
+/****************************************************************************************
+                              GENERIC SEMAPHORE API
+****************************************************************************************/
+
+/*----------------------------------------------------------------
+ *
+ * Function: OS_VxWorks_GenericSemTake
+ *
+ *  Purpose: Local helper routine, not part of OSAL API.
+ *
+ *-----------------------------------------------------------------*/
+int32 OS_VxWorks_GenericSemTake(SEM_ID vxid, int sys_ticks)
+{
+    int vx_status;
+
+    /* Take VxWorks Semaphore */
+    vx_status = semTake(vxid, sys_ticks);
+    if (vx_status != OK)
+    {
+        /*
+         * check for the timeout condition,
+         * which has a different return code and
+         * not necessarily an error of concern.
+         *
+         * vxworks7: if sys_ticks == 0, then if the semaphore can
+         * not be taken S_objLib_OBJ_UNAVAILABLE will be returned
+         */
+        if ((errno == S_objLib_OBJ_TIMEOUT) || (!sys_ticks && (errno == S_objLib_OBJ_UNAVAILABLE)))
```

Comment:
I'm confused about why this file had to be forked ... doesn't vxworks6 have a `S_objLib_OBJ_UNAVAILABLE` symbol as well?  Wouldn't this change apply to both versions?  I would think this is actually a bug in the 6.9 OSAL.

Reply:
i went ahead and  used an #if

Reply:
While I like that we now have it common, the more I'm looking at it, the more I'm failing to see why we need to test for `S_objLib_OBJ_UNAVAILABLE` ...

If called with `sys_ticks` as 0, if the semaphore failed to take for any reason, the proper response up to the app should be `OS_SEM_TIMEOUT`... right?  

Why wouldn't this work:
```
        if (sys_ticks == 0 || errno == S_objLib_OBJ_TIMEOUT)
        {
           return OS_SEM_TIMEOUT;
        }
```